# Purpose

A sample of both 1,300 US-based and 1,300 non-US-based researchers will be sent an intellectual diversity survey. I will randomly select 1,300 records from the US list and 1,300 records from the non-US list. These will serve as the sample first authors for my study. Separate csv files will be generated for each group, which makes creating survey distribution lists for use in Qualtrics much easier.

In [44]:
import pandas as pd


In [45]:
US_dataframe=pd.read_csv('Full_US_Records.csv')
NonUS_dataframe=pd.read_csv('Full_NonUS_Records.csv')

In [4]:
print(len(US_dataframe))
print(len(NonUS_dataframe))

14858
35142


There are 14,858 first authors based in the United States and 35,142 authors based outside of the United States

## Random Sampling: US Based Researchers

In [46]:
US_survey_sample = US_dataframe.sample(n=1300, replace = False)

In [12]:
US_survey_sample.to_csv('US_Survey_Sample.csv')

In [15]:
US_survey_sample

Authors  \
13613                Torregrosa, A; Flint, LE; Flint, AL   
4494   Thorson, JT; Cheng, W; Hermann, AJ; Ianelli, J...   
14049                                       Huettmann, F   
3548   Paudyal, R; Stein, TV; Ober, HK; Swisher, ME; ...   
10831  Fragoso, JMV; Goncalves, F; Oliveira, LFB; Ove...   
...                                                  ...   
2181                       Hart, KM; Guzy, JC; Smith, BJ   
1954   Moore, JD; Andersen, DE; Cooper, T; Duguay, JP...   
4133                         McElligott, MM; Lammers, MO   
12391    Key, MM; Smith, AM; Hanns, B; Kane-Sanderson, P   
13295                               Hawco, NJ; Saito, MA   

                                       Author Full Names  \
13613  Torregrosa, Alicia; Flint, Lorraine E.; Flint,...   
4494   Thorson, James T.; Cheng, Wei; Hermann, Albert...   
14049                                    Huettmann, Falk   
3548   Paudyal, Ramesh; Stein, Taylor V.; Ober, Holly...   
10831  Fragoso, Jose M. V.; Goncalves, Fernando; Oliv...   
...                                                  ...   
2181   Hart, Kristen M.; Guzy, Jacquelyn C.; Smith, B...   
1954   Moore, Joseph D.; Andersen, David E.; Cooper, ...   
4133              McElligott, Megan M.; Lammers, Marc O.   
12391  Key, Marcus M., Jr.; Smith, Abigail M.; Hanns,...   
13295                  Hawco, Nicholas J.; Saito, Mak A.   

                                           Article Title  \
13613  Hydrologic Resilience from Summertime Fog and ...   
4494   Empirical orthogonal function regression: Link...   
14049  The Gods Are Angry: A First-Hand Environmental...   
3548   Recreationists' Perceptions of Scenic Beauty a...   
10831  Visual encounters on line transect surveys und...   
...                                                  ...   
2181   Drivers of realized satellite tracking duratio...   
1954   Migration phenology and patterns of American w...   
4133   Investigating Spinner Dolphin (Stenella longir...   
12391  Rare report of bryozoan fouling of rock lobste...   
13295  Competitive inhibition of cobalt uptake by zin...   

                                            Source Title  \
13613  JOURNAL OF THE AMERICAN WATER RESOURCES ASSOCI...   
4494                               GLOBAL CHANGE BIOLOGY   
14049  HINDU KUSH-HIMALAYA WATERSHEDS DOWNHILL: Lands...   
3548                                             FORESTS   
10831                                           PLOS ONE   
...                                                  ...   
2181                                    MOVEMENT ECOLOGY   
1954                                    WILDLIFE BIOLOGY   
4133                         FRONTIERS IN MARINE SCIENCE   
12391  NEW ZEALAND JOURNAL OF MARINE AND FRESHWATER R...   
13295                         LIMNOLOGY AND OCEANOGRAPHY   

                                         Author Keywords  \
13613  planning; rivers; streams; hydrologic cycle; g...   
4494   delta-correction method; empirical orthogonal ...   
14049                                                NaN   
3548   forest beauty; outdoor recreation; graphic eli...   
10831                                                NaN   
...                                                  ...   
2181   Biologging; Telemetry; Platform terminal trans...   
1954   American woodcock; migration phenology; satell...   
4133   passive acoustic monitoring; spinner dolphin; ...   
12391  Bryozoa; epibiosis; fouling; rock lobsters; Ja...   
13295                                                NaN   

                                           Keywords Plus  \
13613  STREAM TEMPERATURES; LOW CLOUDINESS; CLIMATE-C...   
4494   MARINE FISHES; BERING-SEA; RECRUITMENT; MODEL;...   
14049      POPULATION; CLIMATE; RECRUITMENT; BASIN; BIRD   
3548   RED-COCKADED WOODPECKERS; OUTDOOR RECREATION; ...   
10831  CAMERA-TRAP; MAMMAL POPULATIONS; DENSITIES; WI...   
...                                                  ...   
2181   SEA-TURTLES; CHEL

## Random Sampling: Non US Based Researchers

In [13]:
NonUS_survey_sample = NonUS_dataframe.sample(n=1300, replace = False)

In [14]:
NonUS_survey_sample.to_csv('NonUS_Survey_Sample.csv')

In [16]:
NonUS_survey_sample

Authors  \
18329  Purkayastha, J; Roychoudhury, S; Biswa, BB; Da...   
1847   Augustsson, A; Lennqvist, T; Osbeck, CMG; Tibb...   
8921              Liu, SY; Zhang, MJ; Zhao, YF; Chen, NS   
10666  Sung, YH; Liew, JH; Chan, HK; Lee, WH; Wong, B...   
7148   Penketh, L; Schleimer, A; Labaja, J; Snow, S; ...   
...                                                  ...   
16760  Santini, G; Abolaffio, M; Ossi, F; Franzetti, ...   
4509          Van der Mescht, L; Matthee, S; Matthee, CA   
20484                               Thorn, C; Bogner, FX   
20048  Bhardwaj, M; Erixon, F; Holmberg, I; Seiler, A...   
23511  Shutt-Phillips, K; Pafco, B; Heistermann, M; K...   

                                       Author Full Names  \
18329  Purkayastha, Jayaditya; Roychoudhury, Shubhade...   
1847   Augustsson, A.; Lennqvist, T.; Osbeck, C. M. G...   
8921   Liu, Shuya; Zhang, Mengjia; Zhao, Yongfang; Ch...   
10666  Sung, Yik-Hei; Liew, Jia Huan; Chan, Hoi Kin; ...   
7148   Penketh, Luke; Schleimer, Anna; Labaja, Jessic...   
...                                                  ...   
16760  Santini, Giacomo; Abolaffio, Milo; Ossi, Feder...   
4509   Van der Mescht, Luther; Matthee, Sonja; Matthe...   
20484                 Thorn, Christine; Bogner, Franz X.   
20048  Bhardwaj, Manisha; Erixon, Filippa; Holmberg, ...   
23511  Shutt-Phillips, Kathryn; Pafco, Barbora; Heist...   

                                           Article Title  \
18329  Herpetofaunal Diversity and Conservation Statu...   
1847   Consumption of freshwater fish: A variable but...   
8921   Biodiversity and Spatial-Temporal Dynamics of ...   
10666  Assessing the diet of the endangered Beale's e...   
7148   Scarring patterns of whale sharks,Rhincodon ty...   
...                                                  ...   
16760  Population assessment without individual ident...   
4509   New taxonomic and evolutionary insights releva...   
20484  How Environmental Values Predict Acquisition o...   
20048  Ungulate use of an at-grade fauna passage and ...   
23511  Fecal glucocorticoids and gastrointestinal par...   

                                            Source Title  \
18329  SOCIO-ECONOMIC AND ECO-BIOLOGICAL DIMENSIONS I...   
1847                              ENVIRONMENTAL RESEARCH   
8921   INTERNATIONAL JOURNAL OF ENVIRONMENTAL RESEARC...   
10666  AQUATIC CONSERVATION-MARINE AND FRESHWATER ECO...   
7148   AQUATIC CONSERVATION-MARINE AND FRESHWATER ECO...   
...                                                  ...   
16760                          BASIC AND APPLIED ECOLOGY   
4509               MOLECULAR PHYLOGENETICS AND EVOLUTION   
20484                                     SUSTAINABILITY   
20048                 FRONTIERS IN ENVIRONMENTAL SCIENCE   
23511              GENERAL AND COMPARATIVE ENDOCRINOLOGY   

                                         Author Keywords  \
18329  Herpetofauna; Amphibians; Reptiles; Amchang Wi...   
1847   PFOS; Perfluorinated compounds; Dietary intake...   
8921   Margalefidinium; metabarcoding analysis; Jiaoz...   
10666  Allochthonous resource; Asian turtle crisis; c...   
7148   coastal; endangered species; feeding; fish; Os...   
...                                                  ...   
16760  Camera traps; Abundance estimation; Time-to-ev...   
4509   Taxonomy; Molecular systematics; Siphonaptera;...   
20484  environmental values; nature conservation; cog...   
20048  at-grade fauna passage; animal behavior; road ...   
23511  Primate; Conservation; Endocrine; Parasites; I...   

                                           Keywords Plus  \
18329                                                NaN   
1847   PERFLUOROALKYL ACIDS; ENVIRONMENTAL CONTAMINAN...   
8921   DINOFLAGELLATE COCHLODINIUM-POLYKRIKOIDES; RED...   
10666  FRESH-WATER TURTLES; FATTY-ACIDS; ECOLOGY; STR...   
7148   SHARKS RHINCODON-TYPUS; NINGALOO-MARINE-PARK; ...   
...                                                  ...   
16760  N-MIXTURE MODELS;

# Removing emails that are not the first author's

Just like the author address data, the email data needed to send survey also contains all author emails, not just the first author. I need to grab only the first author emails.

In [17]:
import re
email_regex=r"[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?;?"
#This captures single emails but not when there are multiple
#r'^[a-zA-Z0-9]+(?:\.[a-zA-Z0-9]+)*@[a-zA-Z0-9]+(?:\.[a-zA-Z0-9]+)*$'

In [5]:
US_data=pd.read_csv('US_Survey_Sample.csv')
NonUS_data=pd.read_csv('NonUS_Survey_Sample.csv')

## United States Sample

In [6]:
US_email_list=list(US_data['Email Addresses'])

In [7]:
US_email_strings=[]

for item in US_email_list:  #Make everything in the column  a string of text
    string=str(item)
    US_email_strings.append(string)

In [10]:
US_email_strings

['atorregrosa@usgs.gov',
 'James.Thorson@noaa.gov',
 'fhuettmann@alaska.edu',
 'rpaudyal@ufl.edu; tstein@ufl.edu; holly.ober@ufl.edu; mesw@ufl.edu; ejokela@ufl.edu; dcadams@ufl.edu',
 'fragoso1@mac.com',
 'chrish32@uw.edu',
 'Holly.Ernest@uwyo.edu',
 'nw.fuller@ttu.edu',
 'simon.brandl@ustin.utexas.edu',
 'sfgonski@udel.edu; mhor461@ecy.wa.gov',
 'cperry60@gatech.edu',
 'skupferberg@gmail.com',
 'valerie.pence@cincinnatizoo.org; abby.meyer@bgci.org; jlinsky@atlantabg.org; joachim.gratzfeld@bgci.org; h.pritchard@kew.org; mwestwood@mortonarb.org; ebeckman@mortonarb.org',
 'jmerkle@uwyo.edu',
 'barry.berejikian@noaa.gov',
 'jakesa@nwf.org',
 'arturorv@ucsd.edu',
 'caseycday@gmail.com',
 'Anne.Savage@disney.com',
 'lavallef@berkeley.edu',
 'todd_grant@fws.gov',
 'popescu@ohio.edu',
 'jardine@uw.edu',
 'snbogan@ucsb.edu',
 'devin.cowan@ucf.edu; christina.burton@ucf.edu; William.Moreto@ucf.edu',
 'dn113@humboldt.edu',
 'elg82@cornell.edu; elmg@ucsb.edu',
 'jon.horne@idfg.idaho.gov',
 'kmm015

In [18]:
#finding regex matches
US_match_list=list()

for email in US_email_strings:
    match=re.findall(email_regex,email)
    US_match_list.append(match)


In [19]:
US_match_list

[['atorregrosa@usgs.gov'],
 ['horson@noaa.gov'],
 ['fhuettmann@alaska.edu'],
 ['rpaudyal@ufl.edu;',
  'tstein@ufl.edu;',
  'holly.ober@ufl.edu;',
  'mesw@ufl.edu;',
  'ejokela@ufl.edu;',
  'dcadams@ufl.edu'],
 ['fragoso1@mac.com'],
 ['chrish32@uw.edu'],
 ['rnest@uwyo.edu'],
 ['nw.fuller@ttu.edu'],
 ['simon.brandl@ustin.utexas.edu'],
 ['sfgonski@udel.edu;', 'mhor461@ecy.wa.gov'],
 ['cperry60@gatech.edu'],
 ['skupferberg@gmail.com'],
 ['valerie.pence@cincinnatizoo.org;',
  'abby.meyer@bgci.org;',
  'jlinsky@atlantabg.org;',
  'joachim.gratzfeld@bgci.org;',
  'h.pritchard@kew.org;',
  'mwestwood@mortonarb.org;',
  'ebeckman@mortonarb.org'],
 ['jmerkle@uwyo.edu'],
 ['barry.berejikian@noaa.gov'],
 ['jakesa@nwf.org'],
 ['arturorv@ucsd.edu'],
 ['caseycday@gmail.com'],
 ['avage@disney.com'],
 ['lavallef@berkeley.edu'],
 ['todd_grant@fws.gov'],
 ['popescu@ohio.edu'],
 ['jardine@uw.edu'],
 ['snbogan@ucsb.edu'],
 ['devin.cowan@ucf.edu;', 'christina.burton@ucf.edu;', 'oreto@ucf.edu'],
 ['dn113@hum

In [20]:
#Cleaning the list of regex matches to remove semi colons for multiple entries
list_of_matches=list()

for matches in US_match_list:
    in_list=matches
    out_list= [re.sub(';', '', item) for item in in_list]
    
    list_of_matches.append(out_list)

In [21]:
len(list_of_matches)

1300

In [23]:
list_of_matches

[['atorregrosa@usgs.gov'],
 ['horson@noaa.gov'],
 ['fhuettmann@alaska.edu'],
 ['rpaudyal@ufl.edu',
  'tstein@ufl.edu',
  'holly.ober@ufl.edu',
  'mesw@ufl.edu',
  'ejokela@ufl.edu',
  'dcadams@ufl.edu'],
 ['fragoso1@mac.com'],
 ['chrish32@uw.edu'],
 ['rnest@uwyo.edu'],
 ['nw.fuller@ttu.edu'],
 ['simon.brandl@ustin.utexas.edu'],
 ['sfgonski@udel.edu', 'mhor461@ecy.wa.gov'],
 ['cperry60@gatech.edu'],
 ['skupferberg@gmail.com'],
 ['valerie.pence@cincinnatizoo.org',
  'abby.meyer@bgci.org',
  'jlinsky@atlantabg.org',
  'joachim.gratzfeld@bgci.org',
  'h.pritchard@kew.org',
  'mwestwood@mortonarb.org',
  'ebeckman@mortonarb.org'],
 ['jmerkle@uwyo.edu'],
 ['barry.berejikian@noaa.gov'],
 ['jakesa@nwf.org'],
 ['arturorv@ucsd.edu'],
 ['caseycday@gmail.com'],
 ['avage@disney.com'],
 ['lavallef@berkeley.edu'],
 ['todd_grant@fws.gov'],
 ['popescu@ohio.edu'],
 ['jardine@uw.edu'],
 ['snbogan@ucsb.edu'],
 ['devin.cowan@ucf.edu', 'christina.burton@ucf.edu', 'oreto@ucf.edu'],
 ['dn113@humboldt.edu'],
 

In [29]:
first_author_email=list()
for emails in list_of_matches:
    
     if len(emails)==1:
        first_author_email.append(emails)   
     
     elif len(emails)==0:
        first_author_email.append('NA')
    
     else:
        first_author_email.append(emails[0])


In [30]:
first_author_email

[['atorregrosa@usgs.gov'],
 ['horson@noaa.gov'],
 ['fhuettmann@alaska.edu'],
 'rpaudyal@ufl.edu',
 ['fragoso1@mac.com'],
 ['chrish32@uw.edu'],
 ['rnest@uwyo.edu'],
 ['nw.fuller@ttu.edu'],
 ['simon.brandl@ustin.utexas.edu'],
 'sfgonski@udel.edu',
 ['cperry60@gatech.edu'],
 ['skupferberg@gmail.com'],
 'valerie.pence@cincinnatizoo.org',
 ['jmerkle@uwyo.edu'],
 ['barry.berejikian@noaa.gov'],
 ['jakesa@nwf.org'],
 ['arturorv@ucsd.edu'],
 ['caseycday@gmail.com'],
 ['avage@disney.com'],
 ['lavallef@berkeley.edu'],
 ['todd_grant@fws.gov'],
 ['popescu@ohio.edu'],
 ['jardine@uw.edu'],
 ['snbogan@ucsb.edu'],
 'devin.cowan@ucf.edu',
 ['dn113@humboldt.edu'],
 'elg82@cornell.edu',
 ['jon.horne@idfg.idaho.gov'],
 ['kmm0155@aubum.edu'],
 'clare.doherty.gr@dartmouth.edu',
 ['tttempa@gmail.com'],
 ['heather.christiansen@myfwc.com'],
 ['evans.michael26@gmail.com'],
 ['enrico.pirotta@wsu.edu'],
 ['edontis@usf.edu'],
 ['john.kronenberger@usda.gov'],
 ['jtdeweber@gmail.com'],
 ['julianne_harris@fws.gov'],
 

In [31]:
US_data['First_A_Email']=first_author_email

In [32]:
US_data

Unnamed: 0                                            Authors  \
0          13613                Torregrosa, A; Flint, LE; Flint, AL   
1           4494  Thorson, JT; Cheng, W; Hermann, AJ; Ianelli, J...   
2          14049                                       Huettmann, F   
3           3548  Paudyal, R; Stein, TV; Ober, HK; Swisher, ME; ...   
4          10831  Fragoso, JMV; Goncalves, F; Oliveira, LFB; Ove...   
...          ...                                                ...   
1295        2181                      Hart, KM; Guzy, JC; Smith, BJ   
1296        1954  Moore, JD; Andersen, DE; Cooper, T; Duguay, JP...   
1297        4133                        McElligott, MM; Lammers, MO   
1298       12391    Key, MM; Smith, AM; Hanns, B; Kane-Sanderson, P   
1299       13295                               Hawco, NJ; Saito, MA   

                                      Author Full Names  \
0     Torregrosa, Alicia; Flint, Lorraine E.; Flint,...   
1     Thorson, James T.; Cheng, Wei; Hermann, Albert...   
2                                       Huettmann, Falk   
3     Paudyal, Ramesh; Stein, Taylor V.; Ober, Holly...   
4     Fragoso, Jose M. V.; Goncalves, Fernando; Oliv...   
...                                                 ...   
1295  Hart, Kristen M.; Guzy, Jacquelyn C.; Smith, B...   
1296  Moore, Joseph D.; Andersen, David E.; Cooper, ...   
1297             McElligott, Megan M.; Lammers, Marc O.   
1298  Key, Marcus M., Jr.; Smith, Abigail M.; Hanns,...   
1299                  Hawco, Nicholas J.; Saito, Mak A.   

                                          Article Title  \
0     Hydrologic Resilience from Summertime Fog and ...   
1     Empirical orthogonal function regression: Link...   
2     The Gods Are Angry: A First-Hand Environmental...   
3     Recreationists' Perceptions of Scenic Beauty a...   
4     Visual encounters on line transect surveys und...   
...                                                 ...   
1295  Drivers of realized satellite tracking duratio...   
1296  Migration phenology and patterns of American w...   
1297  Investigating Spinner Dolphin (Stenella longir...   
1298  Rare report of bryozoan fouling of rock lobste...   
1299  Competitive inhibition of cobalt uptake by zin...   

                                           Source Title  \
0     JOURNAL OF THE AMERICAN WATER RESOURCES ASSOCI...   
1                                 GLOBAL CHANGE BIOLOGY   
2     HINDU KUSH-HIMALAYA WATERSHEDS DOWNHILL: Lands...   
3                                               FORESTS   
4                                              PLOS ONE   
...                                                 ...   
1295                                   MOVEMENT ECOLOGY   
1296                                   WILDLIFE BIOLOGY   
1297                        FRONTIERS IN MARINE SCIENCE   
1298  NEW ZEALAND JOURNAL OF MARINE AND FRESHWATER R...   
1299                         LIMNOLOGY AND OCEANOGRAPHY   

                                        Author Keywords  \
0     planning; rivers; streams; hydrologic cycle; g...   
1     delta-correction method; empirical orthogonal ...   
2                                                   NaN   
3     forest beauty; outdoor recreation; graphic eli...   
4                                                   NaN   
...                                                 ...   
1295  Biologging; Telemetry; Platform terminal trans...   
1296  American woodcock; migration phenology; satell...   
1297  passive acoustic monitoring; spinner dolphin; ...   
1298  Bryozoa; epibiosis; fouling; rock lobsters; Ja...   
1299                                                NaN   

                                          Keywords Plus  \
0     STREAM TEMPERATURES; LOW CLOUDINESS; CLIMATE-C...   
1     MARINE FISHES; BERING-SEA; RECRUITMENT; MODEL;...   
2         POPULATION; CLIMATE; RECRUITMENT; BASIN; BIRD   
3     RED-COCKADED WOODPECKERS; OUTDOOR RECREATION; ...   
4     CAMERA-TRAP; MAM

In [33]:
US_data.to_csv('US_Survey_Sample.csv')

## NonUS Sample

In [34]:
NonUS_email_list=list(NonUS_data['Email Addresses'])

In [35]:
NonUS_email_strings=[]

for item in NonUS_email_list:  #Make everything in the column  a string of text
    string=str(item)
    NonUS_email_strings.append(string)

In [36]:
#finding regex matches
NonUS_match_list=list()

for email in NonUS_email_strings:
    match=re.findall(email_regex,email)
    NonUS_match_list.append(match)

In [37]:
#Cleaning the list of regex matches to remove semi colons for multiple entries
list_of_matches=list()

for matches in NonUS_match_list:
    in_list=matches
    out_list= [re.sub(';', '', item) for item in in_list]
    
    list_of_matches.append(out_list)

In [38]:
first_author_email=list()
for emails in list_of_matches:
    
     if len(emails)==1:
        first_author_email.append(emails)   
     
     elif len(emails)==0:
        first_author_email.append('NA')
    
     else:
        first_author_email.append(emails[0])

In [39]:
NonUS_data['First_A_Email']=first_author_email

In [40]:
NonUS_data

Unnamed: 0                                            Authors  \
0          18329  Purkayastha, J; Roychoudhury, S; Biswa, BB; Da...   
1           1847  Augustsson, A; Lennqvist, T; Osbeck, CMG; Tibb...   
2           8921             Liu, SY; Zhang, MJ; Zhao, YF; Chen, NS   
3          10666  Sung, YH; Liew, JH; Chan, HK; Lee, WH; Wong, B...   
4           7148  Penketh, L; Schleimer, A; Labaja, J; Snow, S; ...   
...          ...                                                ...   
1295       16760  Santini, G; Abolaffio, M; Ossi, F; Franzetti, ...   
1296        4509         Van der Mescht, L; Matthee, S; Matthee, CA   
1297       20484                               Thorn, C; Bogner, FX   
1298       20048  Bhardwaj, M; Erixon, F; Holmberg, I; Seiler, A...   
1299       23511  Shutt-Phillips, K; Pafco, B; Heistermann, M; K...   

                                      Author Full Names  \
0     Purkayastha, Jayaditya; Roychoudhury, Shubhade...   
1     Augustsson, A.; Lennqvist, T.; Osbeck, C. M. G...   
2     Liu, Shuya; Zhang, Mengjia; Zhao, Yongfang; Ch...   
3     Sung, Yik-Hei; Liew, Jia Huan; Chan, Hoi Kin; ...   
4     Penketh, Luke; Schleimer, Anna; Labaja, Jessic...   
...                                                 ...   
1295  Santini, Giacomo; Abolaffio, Milo; Ossi, Feder...   
1296  Van der Mescht, Luther; Matthee, Sonja; Matthe...   
1297                 Thorn, Christine; Bogner, Franz X.   
1298  Bhardwaj, Manisha; Erixon, Filippa; Holmberg, ...   
1299  Shutt-Phillips, Kathryn; Pafco, Barbora; Heist...   

                                          Article Title  \
0     Herpetofaunal Diversity and Conservation Statu...   
1     Consumption of freshwater fish: A variable but...   
2     Biodiversity and Spatial-Temporal Dynamics of ...   
3     Assessing the diet of the endangered Beale's e...   
4     Scarring patterns of whale sharks,Rhincodon ty...   
...                                                 ...   
1295  Population assessment without individual ident...   
1296  New taxonomic and evolutionary insights releva...   
1297  How Environmental Values Predict Acquisition o...   
1298  Ungulate use of an at-grade fauna passage and ...   
1299  Fecal glucocorticoids and gastrointestinal par...   

                                           Source Title  \
0     SOCIO-ECONOMIC AND ECO-BIOLOGICAL DIMENSIONS I...   
1                                ENVIRONMENTAL RESEARCH   
2     INTERNATIONAL JOURNAL OF ENVIRONMENTAL RESEARC...   
3     AQUATIC CONSERVATION-MARINE AND FRESHWATER ECO...   
4     AQUATIC CONSERVATION-MARINE AND FRESHWATER ECO...   
...                                                 ...   
1295                          BASIC AND APPLIED ECOLOGY   
1296              MOLECULAR PHYLOGENETICS AND EVOLUTION   
1297                                     SUSTAINABILITY   
1298                 FRONTIERS IN ENVIRONMENTAL SCIENCE   
1299              GENERAL AND COMPARATIVE ENDOCRINOLOGY   

                                        Author Keywords  \
0     Herpetofauna; Amphibians; Reptiles; Amchang Wi...   
1     PFOS; Perfluorinated compounds; Dietary intake...   
2     Margalefidinium; metabarcoding analysis; Jiaoz...   
3     Allochthonous resource; Asian turtle crisis; c...   
4     coastal; endangered species; feeding; fish; Os...   
...                                                 ...   
1295  Camera traps; Abundance estimation; Time-to-ev...   
1296  Taxonomy; Molecular systematics; Siphonaptera;...   
1297  environmental values; nature conservation; cog...   
1298  at-grade fauna passage; animal behavior; road ...   
1299  Primate; Conservation; Endocrine; Parasites; I...   

                                          Keywords Plus  \
0                                                   NaN   
1     PERFLUOROALKYL ACIDS; ENVIRONMENTAL CONTAMINAN...   
2     DINOFLAGELLATE COCHLODINIUM-POLYKRIKOIDES; RED...   
3     FRESH-WATER TURTLES; FATTY-ACIDS; ECOLOGY; STR...   
4     SHARKS RHINCODON

In [41]:
NonUS_data.to_csv('NonUS_Survey_Sample.csv')

# Missing Data

There were 34 records in the US sample and 23 records in the non-US sample that did not have email data. The following code selects replacement records.

In [47]:
full_US_dataframe=pd.read_csv('Full_US_Records.csv')
full_NonUS_dataframe=pd.read_csv('Full_NonUS_Records.csv')

In [48]:
US_sample=pd.read_csv('US_Survey_Sample.csv')
NonUS_sample= pd.read_csv('NonUS_Survey_Sample.csv')

In [49]:

new_US_entries=full_US_dataframe.sample(n=34)


In [51]:
new_US_entries

Authors  \
9151                 Curtis, WJ; Gebhard, AE; Perkin, JS   
192                                          Gompper, ME   
3379   Brook, CE; Herrera, JP; Borgerson, C; Fuller, ...   
8464   Macartney, KJ; Abraham, AC; Slattery, M; Lesse...   
14415  Finger, N; Farleigh, K; Bracken, JT; Leache, A...   
13891                              Love, JW; Newhard, JJ   
12527  Howard, JL; Creed, JC; Aguiar, MVP; Fouqurean, JW   
12738              Li, WT; Chiang, YL; Chen, TY; Lai, CL   
6988                           McClanahan, TR; Azali, MK   
12571  McHuron, EA; Luxa, K; Pelland, NA; Holsman, K;...   
1631   Garabedian, JE; Moorman, CE; Peterson, MN; Kil...   
14075                    Hartin, JS; Surls, RA; Bush, JP   
11257  McCarter, J; Sterling, EJ; Jupiter, SD; Cullma...   
472      Aoki, LR; McGlathery, KJ; Wiberg, PL; Al-Haj, A   
13226                       Goldenberg, SZ; Wittemyer, G   
2712                                         Eliason, SL   
6042   Howell, PE; Muths, E; Hossack, BR; Sigafus, BH...   
13895  Elkins, D; Sweat, SC; Kuhajda, BR; George, AL;...   
3446                             Dickson, BV; Pierce, SE   
10060                                 Erb, A; Turner, AH   
2824   Barnett, KM; Detmering, SE; McMahon, TA; Civit...   
1677                          Triezenberg, HA; Knuth, BA   
5567   Evans, N; Schroeder, ID; Buil, MP; Jacox, MG; ...   
13311  Higashi, CHV; Nichols, WL; Chevignon, G; Patel...   
11469   Roberts, CP; Twidwell, D; Angeler, DG; Allen, CR   
8130   Ocon, JP; Ibanez, T; Franklin, J; Pau, S; Kepp...   
12539            Shaver, EC; Burkepile, DE; Silliman, BR   
569     Ruscher, B; Sills, JM; Richter, BP; Reichmuth, C   
6188   Murphy, SM; Vidal, MC; Smith, TP; Hallagan, CJ...   
5283   Wilkins, EJ; Cole, NW; Miller, HM; Schuster, R...   
13336                      West, EH; Brunk, K; Peery, MZ   
4783                                          Lockley, M   
14658  Fukunaga, A; Pascoe, KH; Pugh, AR; Kosaki, RK;...   
14387  Graham, AM; Lavretsky, P; Munoz-Fuentes, V; Gr...   

                                       Author Full Names  \
9151   Curtis, William J.; Gebhard, Amy E.; Perkin, J...   
192                                  Gompper, Matthew E.   
3379   Brook, Cara E.; Herrera, James P.; Borgerson, ...   
8464   Macartney, Keir J.; Clayshulte Abraham, Amelia...   
14415  Finger, Nicholas; Farleigh, Keaka; Bracken, Ja...   
13891                Love, Joseph W.; Newhard, Joshua J.   
12527  Howard, Jason L.; Creed, Joel C.; Aguiar, Mari...   
12738  Li, Wen-Ta; Chiang, Yi-Lun; Chen, Ting-Yu; Lai...   
6988        McClanahan, Timothy R.; Azali, Maxwell Kodia   
12571  McHuron, Elizabeth A.; Luxa, Katie; Pelland, N...   
1631   Garabedian, James E.; Moorman, Christopher E.;...   
14075  Hartin, Janet S.; Surls, Rachel A.; Bush, Jose...   
11257  McCarter, Joe; Sterling, Eleanor J.; Jupiter, ...   
472    Aoki, Lillian R.; McGlathery, Karen J.; Wiberg...   
13226           Goldenberg, Shifra Z.; Wittemyer, George   
2712                                 Eliason, Stephen L.   
6042   Howell, Paige E.; Muths, Erin; Hossack, Blake ...   
13895  Elkins, Duncan; Sweat, Sarah C.; Kuhajda, Bern...   
3446             Dickson, Blake, V; Pierce, Stephanie E.   
10060                       Erb, Arthur; Turner, Alan H.   
2824   Barnett, K. M.; Detmering, S. E.; McMahon, T. ...   
1677          Triezenberg, Heather A.; Knuth, Barbara A.   
5567   Evans, Natalya; Schroeder, Isaac D.; Pozo Buil...   
13311  Higashi, Clesson H. V.; Nichols, William L. L....   
11469  Roberts, Caleb P.; Twidwell, Dirac; Angeler, D...   
8130   Ocon, Jonathan Pando; Ibanez, Thomas; Franklin...   
12539  Shaver, Elizabeth C.; Burkepile, Deron E.; Sil...   
569    Ruscher, Brandi; Sills, Jillian M.; Richter, B...   
6188   Murphy, Shannon M.; Vidal, Mayra C.; Smith, Ti...   
5283   Wilkins, Emily J.; Cole, Nicholas W.; Miller, ...   
13336  West, Elena H.; Brunk, Kristin; Peery, M. Zach...   
4783                        

In [53]:
new_US_entries['Email Addresses']
#Emails that were not duplicates were added to the distribution list

9151     mrlliw1@gmail.com; amy.e.doll1@gmail.com; jper...
192                                      gompperm@nmsu.edu
3379                                   cbrook@berkeley.edu
8464                              kjm1049@wildcats.unh.edu
14415                             cblair@citytech.cuny.edu
13891                             joseph.love@maryland.gov
12527                                     jhowa033@fiu.edu
12738                               heerolee1104@gmail.com
6988                                   tmcclanahan@wcs.org
12571                                      emchuron@uw.edu
1631                                    jegarabe@gmail.com
14075                                   jshartin@ucanr.edu
11257                                                  NaN
472                                      lra53@cornell.edu
13226                        shifra.z.goldenberg@gmail.com
2712                              seliason@msubillings.edu
6042                              paige.e.howell@gmail.c

In [61]:
new_NonUS_entries=full_NonUS_dataframe.sample(n=23)

In [62]:
new_NonUS_entries['Email Addresses']

1619                               leon.green@bioenv.gu.se
29824                                    wrrnovais@uesc.br
32862                      sasha.whitmarsh@flinders.edu.au
5381                                klaus.glenk@sruc.ac.uk
29638                                  mvcondini@gmail.com
23755                              y.chen.25@warwick.ac.uk
11974                                       hillera@si.edu
31395                           dustin.marshall@monash.edu
5429                                   almalki.m@tu.edu.sa
14356           julialesherl@gmail.com; smachkou@ecosur.mx
8090                                    xiaoshuhuo@163.com
31533    hector.barriosgarrido@my.jcu.edu.au; jordanopa...
13822    npkoju.2003@gmail.com; rckandel06@gmail.com; h...
5190         gpeniche@hotmail.com; vikingrobin@hotmail.com
1289     kwyen@mail.tfrin.gov.tw; cjpan@mail.tfrin.gov....
2178                             imiqueleiz@alumni.unav.es
16767        miriam.ruocco@szn.it; emanuela.dattolo@szn.

In [66]:
s=full_NonUS_dataframe.sample(n=3)
s['Email Addresses']

18387    v.subramanian@qu.edu.qa
13305            Maren142@web.de
2643         mturon@ceab.csic.es
Name: Email Addresses, dtype: object